core/src/main/kotlin/cache/data/MessageInteractionData.kt

```diff
@@ -0,0 +1,24 @@
+package cache.data;
+
+import dev.kord.common.annotation.KordPreview
+import dev.kord.common.entity.DiscordMessageInteraction
+import dev.kord.common.entity.InteractionType
+import dev.kord.common.entity.Snowflake
+import dev.kord.core.cache.data.UserData
+import dev.kord.core.cache.data.toData
+import kotlinx.serialization.Serializable
+
+@KordPreview
+@Serializable
+data class MessageInteractionData(
+    val id:Snowflake,
+    val type:InteractionType,
+    val name:String,
+    val user:UserData
```

Comment:
Flatten this out to the user id

Reply:
I applied the user data based on #147.
But  I guess that's not the scope for this PR will turn this into the user id.